src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/GridGeometryProvider.java

```diff
@@ -264,6 +265,44 @@ private void transformResolution(
                 resolution[1] = transformedDY;
             }
         }
+
+        /** Gets granules' resolution if it is equal for all the granules, otherwise returns null */
+        public double[] getGranulesNativeResolutionIfSame(GranuleSource granuleSource)
+                throws IOException {
+            Map<String, DimensionDescriptor> descriptors = crsRequestHandler.getDescriptors();
+            DimensionDescriptor resDescriptor = descriptors.get(DimensionDescriptor.RESOLUTION);
+            DimensionDescriptor resXDescriptor = descriptors.get(DimensionDescriptor.RESOLUTION_X);
+            DimensionDescriptor resYDescriptor = descriptors.get(DimensionDescriptor.RESOLUTION_Y);
+            final String resXAttribute =
+                    hasBothResolutions
+                            ? resXDescriptor.getStartAttribute()
+                            : resDescriptor.getStartAttribute();
+            final String resYAttribute =
+                    hasBothResolutions
+                            ? resYDescriptor.getStartAttribute()
+                            : resDescriptor.getStartAttribute();
+
+            SimpleFeatureCollection granules = granuleSource.getGranules(Query.ALL);
```

Comment:
Be careful.
This query will return ALL the granules from the DB. They might be thousands/millions for a whole country.
I think that we should refine the query using a bbox or the ROI or something similar as we did in other methods around.